plugins/bazaar-backend/src/service/router.ts

```diff
@@ -19,30 +19,73 @@ import express from 'express';
 import Router from 'express-promise-router';
 import { Logger } from 'winston';
 import { Config } from '@backstage/config';
-import { IdentityApi } from '@backstage/plugin-auth-node';
+import {
+  getBearerTokenFromAuthorizationHeader,
+  IdentityApi,
+} from '@backstage/plugin-auth-node';
 import { DatabaseHandler } from './DatabaseHandler';
+import { NotAllowedError } from '@backstage/errors';
+import {
+  PermissionEvaluator,
+  AuthorizeResult,
+  BasicPermission,
+} from '@backstage/plugin-permission-common';
+import { createPermissionIntegrationRouter } from '@backstage/plugin-permission-node';
+// eslint-disable-next-line @backstage/no-relative-monorepo-imports
+import {
+  bazaarAddPermission,
+  bazaarDeletePermission,
+  bazaarUpdatePermission,
+  bazaarPermissions,
+} from '@backstage/plugin-bazaar-common';
 
 /** @public */
 export interface RouterOptions {
   logger: Logger;
   database: PluginDatabaseManager;
   config: Config;
   identity: IdentityApi;
+  permissions: PermissionEvaluator;
```

Comment:
this should be optional to avoid breaking changes for existing users

```suggestion
  permissions?: PermissionEvaluator;
```